### PR TITLE
M255 Deployment Smoke

### PR DESCRIPTION
**Pull request by Ollieollieolllie**
This got lost somewhere along the way

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+Examples:
+
+Author:
+    Snow(Dryden)
+---------------------------------------------------------------------------- */
+params ["_projectile", "_submunitionProjectile", "_pos", "_velocity"];
+
+if (typeOf _submunitionProjectile == "fza_flec_cluster") then {
+    _counter = floor random 100;
+    if (_counter > 3) exitWith {};
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf:26]: Has this been tested in multiplayer? My understanding was that `SubmunitionCreated` only ran on a single machine, and `drop` only has local effect

File: @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* ----------------------------------------------------------------------------
+Function: fza_fnc_eventFired
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf:2]: This looks incorrect?

File: @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+Description:
+    Submunition creation event handler
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf:5]: The description should help explain what this function does

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf [hunk 4]
```diff
@@ -122,7 +122,7 @@ fza_ah64_firekeypressed = 0;
 fza_ah64_overallticker = 0;
 fza_ah64_locktargstate = 0;
 fza_ah64_irjammer = 0;
-fza_ah64_salvofired = 0;
+fza_ah64_salvofired = 1;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/XEH_preInit.sqf:125]: I don't think this is the correct way. To me it looks like the fired event handler order is incorrect. Maybe try in `extendedEventHandlers.hpp` changing the fired eventhandler from

```cpp
firedBIS = "_this call fza_fnc_eventFired; _this call fza_fnc_fxMuzzle; fza_ah64_salvofired = fza_ah64_salvofired + 1;";
```

to

```cpp
firedBIS = " _this call fza_fnc_fxMuzzle; fza_ah64_salvofired = fza_ah64_salvofired + 1;_this call fza_fnc_eventFired;";
```

File: @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    Nothing
+
+Examples:
+
+Author:
+    Snow(Dryden)
+---------------------------------------------------------------------------- */
+params ["_projectile", "_submunitionProjectile", "_pos", "_velocity"];
+
+if (typeOf _submunitionProjectile == "fza_flec_cluster") then {
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_hydra/functions/fn_eventFired.sqf:24]: Could you early-exit here?

```sqf
if (typeOf _submunitionProjectile != "fza_flex_cluster") exitWith {}
```

**Review by mattysmith22 — CHANGES_REQUESTED** [review]

**Review by BradMick — APPROVED** [review]
Good to go

**Review by mattysmith22 — APPROVED** [review]

**Pull request merged**